apis/v1beta1/httproute_types.go

```diff
@@ -901,7 +901,8 @@ type HTTPRequestRedirectFilter struct {
 
 	// Port is the port to be used in the value of the `Location`
 	// header in the response.
-	// When empty, port (if specified) of the request is used.
+	// When empty, port (if specified) in the `Host` header
```

Comment:
Copying my comment from discussion here, I would push for this to be the parent (gateway listener) port personally. 

On a complex multi-hop network path, there is no way we can know the "right" port to use. IMO, using the most stable thing we have - parent port - is a reasonable compromise. We should indicate to users that setting port explicitly is a best practice in these scenarios.

I don't feel too strongly if everyone else likes Host and envoy implements it.

Reply:
when the `hostname` setting is empty, the hostname from the `Host` header is used, so imho
when the `port` setting is empty, the port from the `Host` header should be used :)

Reply:
regarding the current implementation limitation - this should be fixed in envoy soon https://github.com/envoyproxy/envoy/pull/25573 (thanks @cpakulski ! )

Reply:
Here is some prior art in the "L4" space:

```
$ curl -s wikipedia.org -H "Host: wikipedia.org:8080" -v |& grep Location
< Location: https://wikipedia.org/
$ curl -s google.com -H "Host: google.com:8080" -v |& grep Location
< Location: http://www.google.com/
$ curl -s gateway-api.org -H "Host: gateway-api.org:8080" -v |& grep Location
< Location: http://gateway-api.sigs.k8s.io
$ curl httpbin.org/relative-redirect/1 -H "Host: httpbin.org:8080" -v -s |& grep Location
< Location: /get
$ curl httpbin.org/redirect-to?url=http://a -H "Host: httpbin.org:8080" -v -s |& grep Location
< Location: http://a
```

No one cares about the port in Host that I have found.

Reply:
I tend to agree with @howardjohn here. As far as I can tell, defaulting to the port of the Gateway listener (or Service for GAMMA) seems like the most broadly implementable solution. The current spec is unfortunately too vague, so a big thanks for @arkodg for highlighting this and working to clarify the spec.

Reply:
thanks for running these tests, if we go with above, then the default semantics should be that the Location header contains the stripped hostname (no port) when port redirect is unset

Reply:
@robscott from the current docs, its much easier to imply that the original author was referring to L7 Host Header due to this phrase `port (if specified)`, hence this PR simply makes it more clear
any other change makes it an API change (which is fine, but needs more scrutiny)

Reply:
are you suggesting that when `port-redirect` is unset, and the Host header contains `example.com` but the `port` within the `listener` in the `Gateway` resource is set to `10080`, the Location header should be `example.com:10080` ?
this conflicts with https://github.com/kubernetes-sigs/gateway-api/pull/1740#discussion_r1113268023 as well as the suggested PR change

Reply:
also would be great if @hbagdi ([author](https://github.com/kubernetes-sigs/gateway-api/pull/683)) can help clarify the default semantics

Reply:
So far, I do believe that's what's being suggested by John, Rob and myself. I can be persuaded, but the listenerPort fallback does seem more broadly implementable.

Reply:
updated the language to point to listener port, thanks for the discussion !